src/js/yeldaWebchatInjector.js

```diff
@@ -976,6 +997,34 @@ class YeldaChat {
     }
   }
 
+  /**
+   * Listen to url update on the page
+   * @param {Boolean} shouldRemoveEvent
+   */
+  listenUrlUpdate(shouldRemoveEvent = false) {
+    const eventHandler = window[shouldRemoveEvent ? 'removeEventListener' : 'addEventListener']
+
+    if (!this.urlUpdateListenerBind) {
+      this.urlUpdateListenerBind = this.urlUpdateListener.bind(this)
+    }
+    // mouseout event is used to listen leave viewport
+    eventHandler('locationchange', this.urlUpdateListenerBind)
+    eventHandler('popstate', this.urlUpdateListenerBind)
+    eventHandler('hashchange', this.urlUpdateListenerBind)
```

Comment:
i'll accept that PR, but the code is not enough : these event are not triggered on PWA 
Try adding the webchat on yelda and see if url is updated when the user navigate .. it does not.
i took your code and added console.log, it does not work on app navigation, only if i click on browser back button

![Selection_253](https://user-images.githubusercontent.com/5340081/138089227-21b741e3-724c-480a-b330-324c9749737a.png)
.